script.js

```diff
@@ -1,4 +1,181 @@
+var randomChoice = function () {
+  var randomDecimal = Math.random() * 3;
+  var randomNumber = Math.floor(randomDecimal);
+  var choiceNumber = randomNumber + 1;
+  return choiceNumber;
+};
+
+var randomDiceRoll = function () {
+  var randomDec = Math.random() * 6;
+  var randomNum = Math.floor(randomDec);
+  var randomDice = randomNum + 1;
+  return randomDice;
+};
+
+var currentGameMode = "waiting for user name";
+
+var userRecord = 0;
+
+var programRecord = 0;
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  // input = scissors, paper or stone
+  // output = user won, program won or it's a draw
+  var choiceNumber = randomChoice();
+
+  var myOutputValue;
+
+  var inputNumber;
+
+  var choice;
+
+  if (currentGameMode == "waiting for user name") {
+    //set user name as input
+    userName = input;
+
+    currentGameMode = "waiting for choice of game";
+
+    return (
+      "Hello " +
+      userName +
+      " Please type in Word Game, Dice Roll or leave blank for Scissors Paper Stone Game."
+    );
+  }
+
+  if (currentGameMode == "waiting for choice of game") {
+    console.log("banana");
+    if (input == "Word Game") {
+      myOutputValue = "Guess the secret word!";
+      currentGameMode = "Secret Word";
+      return myOutputValue;
+    }
+
+    if (input == "") {
+      currentGameMode = "Scissors Paper Stone";
+      myOutputValue = "We will proceed with the Scissors Paper Stone Game.";
+      return myOutputValue;
+    }
+
+    if (input == "Dice Roll") {
+      currentGameMode = "Dice Roll Game";
+      myOutputValue =
+        "We will proceed with Dice Roll Game. Please input your guess between 1 to 6!";
+      return myOutputValue;
+    }
+  }
+  // if Game Mode is Dice Roll Game, we will play guess the dice num
+  if ((currentGameMode = "Dice Roll Game")) {
+    var diceRoll = randomDiceRoll();
+    myOutputValue = "Wrong! Guess again!";
+    if (input == diceRoll) {
+      myOutputValue = "You guessed the right number!";
+    }
+    return myOutputValue;
+  }
+
+  // if Game mode is Secret Word, we will play guess the secret word
+  if (currentGameMode == "Secret Word") {
+    myOutputValue = "Wrong! Try again";
+    if (input == "banana") {
+      myOutputValue = "You guessed the secret word!";
+    }
+    return myOutputValue;
+  }
+
```

Comment:
Nice how you combined all games into one file, i would suggest to abstract different game logic into thier own separate functions and call them in your main function when appropiate! 👍